sympy/physics/control/control_plots.py

```diff
@@ -731,23 +733,33 @@ def bode_magnitude_numerical_data(system, initial_exp=-5, final_exp=5, **kwargs)
 
     mag = 20*log(Abs(w_expr), 10)
 
-    return LineOver1DRangeSeries(mag,
+    x, y = LineOver1DRangeSeries(mag,
         (_w, 10**initial_exp, 10**final_exp), xscale='log', **kwargs).get_points()
+    if freq_unit == 'Hz':
```

Comment:
Move this to `w_expr` on line 732.

```
if freq_unit == 'Hz':
    repl = I*_w/(2*pi)
else:
    repl = I*_w
w_expr = expr.subs({system.var: repl})
```

and the same for phase (for `phase_unit` you can multiply `phase = arg(w_expr)` with `180/pi` in case of degrees *before*  evaluating it).

Probably it also makes sense moving the check of valid units to the numerical_data-functions.

Reply:
Thanks for this .Even time wise this solution is better .For `phase = arg(w_expr)` multiplying works fine and gives the expected graphs .While for `repl = I*_w/(2*pi) ` there is an observation on applying the diff you suggested .

By `repl = I*_w/(2*pi)` x axis values are multiplied by 2*pi (instead of divided by it which is expected)


![OG2](https://user-images.githubusercontent.com/90216905/154244763-34758939-9dd5-404f-9a3a-0522836a2632.png)

Having seen this it was obvious to try the inverse operation.
By `repl = I*_w*(2*pi)` x axis values are divided by 2*pi ( which gives the expected plots )

![OG1](https://user-images.githubusercontent.com/90216905/154243584-42119c09-652d-43de-83aa-bad831798cbe.png)

This needs to be investigated why` I*_w*(2*pi)` gives the correct results and should we go forward with that ?

Reply:
@akshanshbhatt Any thoughts on this ?

Reply:
CC @oscargus @akshanshbhatt I have made all the other changes ,would like if you addressed this comment .Both the current and this suggested code run in similar time ( the suggestion a bit faster ) . Other than that the plots are identical and as expected !

Reply:
Go ahead with the suggested one. It looks better and is more pythonic.

Reply:
` I*_w*(2*pi)` is the correct version, my bad.

Reply:
The point is to use the symbolic functionality rather than floating-point computations.

Reply:
Thanks @oscargus @akshanshbhatt .I have made all the above changes in the latest commit .It seems that the `sphinx.sh`
check will fail (I guess because of the change `mag.xlabel(None)` ) and I'm not too familiar with it .I will look into that today !